src/components/CatagoryList/CatagoryList.tsx

```diff
@@ -63,40 +101,131 @@ export default function BasicTable() {
     });
     setOpen(false);
   };
+
+  const handleSort = (name: string) => {
+    const isAsc = orderBy === name && order === "asc";
+    setOrder(isAsc ? "desc" : "asc");
+    setOrderBy(name);
+
+    if (orderBy === "category") {
+      setData(
+        data.sort((a, b) => {
```

Comment:
* Sort actually mutate the original array -> this means it'll mutate the state which results in unexpected bugs.
* It's a bad idea to store computed state that you can compute on render from other states